### DISCUSSION TITLE
Please share more details about IResourceManager & ResourceManagerRequested

### DISCUSSION THREAD

**Discussion by Jack251970** [microsoft/WindowsAppSDK#4805] [Q&A]
[Office document](https://learn.microsoft.com/zh-cn/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.application.resourcemanagerrequested?view=windows-app-sdk-1.6) has provided these sample codes:
```
ResourceManagerRequested += (_, e) =>
 {
    // CreateResourceManager() is a custom method you
    // create that returns an instance of IResourceManager.
    IResourceManager resourceManager = CreateResourceManager();
    e.ResourceManager = resourceManager;
 };
```
But I am still confused about how to get a **custom** `IResourceManager` instance.

`IResourceManager` has one property related to the resources only - `ResourceMap MainResourceMap`.

For `ResourceMap`, it is a sealed class without public function to add & delete its resources. Similarly, `ResourceManager` is also a sealed class and can only load one resource file. It seems that I cannot manage its resources and the `IResourceManager` instance I can get is either the default application resources or the resources in one resource file (from `ResourceManager`).

Now, I want to get a `IResourceManager` instance, which contains not only the default application resources but also the resources in one resource file. How can I do that?

**Comment by castorix** [comment]
Theorically it should be by implementing a class like 
_public class CustomResourceManager : IResourceManager_
by adding mainly
_public ResourceMap MainResourceMap_ (with get{}, it is called many times at beginning if I put a BP (I returned __resourceManager.MainResourceMap_))
and 
 _ResourceContext IResourceManager.CreateResourceContext()_
where I tried to test by overriding the scale like:
```
private ResourceManager _resourceManager;
```
in constructor :
` _resourceManager = new ResourceManager();`
then in CreateResourceContext
```
 var resourceContext = _resourceManager.CreateResourceContext();
resourceContext.QualifierValues["Scale"] = "200";
 return resourceContext;
```
but it changed nothing...
